nep-X1.mediawiki

```diff
@@ -0,0 +1,60 @@
+<pre>
+  NEP: 31
+  Title: Contract Destroy Function
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>
+  Type: Standard
+  Status: Draft
+  Created: 2024-09-11
+</pre>
+
+==Abstract==
+
+This proposal standardizes the destroy function which is necessary for contract storage deletion.
+
+==Motivation==
+
+N3 already provided contract destroy mechanism but it hasn't been summarized in a standard. This standard will show details.
+
+==Specification==
+
+Neo N3 has the native <code>ContractManagement</code> contract that is responsible for contract destruction via its <code>destroy</code> method. When <code>destroy</code> is executed, it will delete all storage entries of this contract. It will block the contract address using the native <code>Policy</code> contract to prevent redeployment or calling any methods on the contract from that point forward. The user smart contract should implement the <code>destroy</code> method as described below.
+
+===Methods===
+
+====destroy====
+
+<pre>
+{
+  "name": "destroy",
+  "safe": false,
+  "parameters": [],
+  "returntype": "Void"
+}
+</pre>
+This method MUST invoke the <code>destroy</code> method of <code>ContractManagement</code> contract when the contract is destroyed.
```

Comment:
It's a subject for discussion, but I'd suggest to specify native Management's Destroy signature and necessary callflags in this standard, because it's something that's written in stones for N3.